storage/innobase/srv/srv0srv.cc

```diff
@@ -1421,6 +1422,31 @@ srv_export_innodb_status(void)
 		btr_scrub_total_stat(&scrub_stat);
 	}
 
+#ifdef BTR_CUR_HASH_ADAPT
+	mem_adaptive_hash = 0;
+	ut_ad(btr_search_sys->hash_tables);
+	for (ulong i = 0; i < btr_ahi_parts; i++) {
+		hash_table_t*	ht = btr_search_sys->hash_tables[i];
+
+		ut_ad(ht);
+		ut_ad(ht->heap);
+		/* Multiple mutexes/heaps are currently never used for adaptive
+		hash index tables. */
+		ut_ad(!ht->n_sync_obj);
+		ut_ad(!ht->heaps);
+
+		mem_adaptive_hash += mem_heap_get_size(ht->heap);
+		mem_adaptive_hash += ht->n_cells * sizeof(hash_cell_t);
+	}
+
+	export_vars.innodb_mem_adaptive_hash = mem_adaptive_hash;
+	export_vars.innodb_adaptive_hash_hash_searches = btr_cur_n_sea;
+	export_vars.innodb_adaptive_hash_non_hash_searches = btr_cur_n_non_sea;
+#endif
+	mem_dictionary = (dict_sys.table_hash->n_cells + 
```

Comment:
I don’t think that we need the local variable `mem_dictionary`. We can assign directly to `export_vars.innodb_mem_dictionary` and avoid the `ulint`/`ulong` type mismatch.

Reply:
As for the local `mem_adaptive_hash`, should we remove that also and directly modify `export_vars.innodb_mem_adaptive_hash`?

Reply:
Also, for `export_vars.innodb_mem_dictionary`, should we be holding `dict_sys.mutex`?